crates/ibc/src/core/ics26_routing/handler.rs

```diff
@@ -44,7 +42,7 @@ where
 }
 
 /// Attempts to convert a message into a [MsgEnvelope] message
-pub fn decode(message: Any) -> Result<MsgEnvelope, RouterError> {
+pub(crate) fn decode(message: Any) -> Result<MsgEnvelope, RouterError> {
```

Comment:
Let's leave this one `pub`, as there might be use cases for it (see [basecoin](https://github.com/informalsystems/basecoin-rs/blob/c5744f4a1eac9a63ef481410e52d9fb40363b97e/src/app/modules/ibc.rs#L298))

Reply:
Cancel that, I'll post a new solution soon